processing/src/main/java/io/druid/query/search/SearchQueryRunner.java

```diff
@@ -102,16 +116,46 @@ public SearchQueryRunner(Segment segment)
     // Closing this will cause segfaults in unit tests.
     final QueryableIndex index = segment.asQueryableIndex();
 
+    final StorageAdapter storageAdapter = segment.asStorageAdapter();
+
+    // Split dimension list into bitmap-supporting list and non-bitmap supporting list
+    Iterable<DimensionSpec> dimsToSearch;
+    if (dimensions == null || dimensions.isEmpty()) {
+      dimsToSearch = Iterables.transform(storageAdapter.getAvailableDimensions(), Druids.DIMENSION_IDENTITY);
+    } else {
+      dimsToSearch = dimensions;
+    }
+
+    final List<DimensionSpec> bitmapDims;
+    final List<DimensionSpec> nonbitmapDims;
     if (index != null) {
-      final TreeMap<SearchHit, MutableInt> retVal = Maps.newTreeMap(query.getSort().getComparator());
+      bitmapDims = Lists.newArrayList();
+      nonbitmapDims = Lists.newArrayList();
+      for (DimensionSpec spec : dimsToSearch) {
+        if (spec.getDimension().equals(Column.TIME_COLUMN_NAME)) {
+          bitmapDims.add(spec);
+          continue;
+        }
+        ColumnCapabilities capabilities = storageAdapter.getColumnCapabilities(spec.getDimension());
+        if (capabilities == null) {
+          continue;
```

Comment:
If capabilities is null, the column doesn't exist, so I thought there would be little value in performing a search against it